inc/Engine/Media/Fonts/Frontend/Controller.php

```diff
@@ -244,6 +252,50 @@ public function disable_google_fonts_preload( $disable ): bool {
 		return true;
 	}
 
+	/**
+	 * Get the list of patterns to exclude from media fonts rewrite.
+	 *
+	 * @return string[]
+	 */
+	private function get_exclusions(): array {
+		/**
+		 * Filters the list of patterns to exclude from media font rewrite.
+		 *
+		 * @since 3.18
+		 *
+		 * @param string[] $exclusions The list of patterns to exclude from media fonts.
+		 */
+		return wpm_apply_filters_typed( 'string[]', 'rocket_exclude_locally_host_fonts', [] );
+	}
+
+	/**
+	 * Checks if a URL is excluded based on the provided exclusions.
+	 *
+	 * @param string   $url        The URL to check.
+	 * @param string[] $exclusions The list of exclusions.
+	 *
+	 * @return bool True if the URL is excluded, false otherwise.
+	 */
+	private function is_excluded( string $url, array $exclusions ): bool {
+		// Bail out early if there are no exclusions.
+		if ( empty( $exclusions ) ) {
+			return false;
+		}
+
+		// Escape each exclusion pattern to prevent regex issues.
+		$escaped_exclusions = array_map(
+				function ( $exclusion ) {
+					return preg_quote( $exclusion, '#' ); // '#' is used as the delimiter.
```

Comment:
Thinking back on this, preg_quote() is going to prevent RegEx characters from doing what we need (as reported by Hanna), so the escaping should only be done for the `#` character, using a `str_replace( '#', '\#', string )` for example